lib/applications.js

```diff
@@ -1,15 +1,26 @@
 'use strict';
 
+const semver = require('semver');
+
 const errors = require('./errors');
 
 // Generic application class. You are free to substitute it with whatever suits
 // your needs.
-//   name - application name
-//   api - application API
+//   name - application name that may contain version after '@'
+//          (e.g. app@1.0.0). Version in name is preferred over
+//          'version' parameter
+//   api - object that contains interfaces as its fields each of which
+//         contains functions by method names. Each method has the following
+//         signature (connection, <0 or more method arguments>, callback)
+//   version - application version of 'name' application
 //
 class Application {
-  constructor(name, api) {
-    this.name = name;
+  constructor(name, api, version) {
+    [this.name, this.version] = name.split('@');
```

Comment:
@aqrln, by the way, are you sure that `@metarhia/application@1.0` is a valid package name?
Trying to create the package with such name using `npm` leads to the error:
`Sorry, name can only contain URL-friendly characters.`